lib/active_model_serializers/adapter/json_api/resource_identifier.rb

```diff
@@ -20,7 +20,8 @@ def as_json
         private
 
         def type_for(serializer)
-          return serializer._type if serializer._type
+          return serializer._type.call(serializer.object) if serializer._type.respond_to?(:call)
```

Comment:
I think using `serializer.instance_eval(serializer._type)` would be more appropriate for instance if you want to use a function defined on the serializer or something similar and is closer to the other block usage we have.